src/stampy_nlp/search.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import urllib
+from stampy_nlp.utilities.pinecone_utils import DEFAULT_TOPK, query as query_db
+from stampy_nlp.utilities.huggingface_utils import get_reader, get_retriever, lit_search as hf_search
+from stampy_nlp.utilities.coda_utils import LIVE_STATUS
+from stampy_nlp.faq_titles import encode_faq_titles
+
+logger = logging.getLogger(__name__, )
+logger.setLevel(logging.DEBUG)
+
+COUNT: int = 3
+DEFAULT_QUERY: str = 'What is AI Safety?'
+BLANK: str = ''
+ALL: str = 'all'
+
+
+def semantic_search(query: str, top_k: int = DEFAULT_TOPK, showLive: bool = True, status=[], namespace: str = 'faq-titles'):
+    """Semantic search for query"""
+    logger.info("semantic_search: %s", query)
+    logger.debug("params top_k=%s, status=%s, showLive=%s", top_k, status, showLive)
```

Comment:
loggers should use this formatting, as they're lazy evaluated, and so only do the string interpolation if its needed. This is seldom a problem, but sometimes debugging info can contain large (or slow) operations